pkg/manifests/config.go

```diff
@@ -358,19 +338,27 @@ func UnmarshalStrict(data []byte, v interface{}) error {
 	}
 }
 
-func newConfig(content []byte, collectionProfilesFeatureGateEnabled bool) (*Config, error) {
+func newConfig(content []byte, collectionProfilesFeatureGateEnabled bool) (*Config, *InvalidConfigWarning, error) {
 	c := Config{CollectionProfilesFeatureGateEnabled: collectionProfilesFeatureGateEnabled}
+
+	var warning *InvalidConfigWarning
+	wCmc := defaultClusterMonitoringConfiguration()
+	wErr := UnmarshalStrict(content, &wCmc)
+	if wErr != nil {
+		warning = &InvalidConfigWarning{ConfigMap: "openshift-monitoring/cluster-monitoring-config", Err: wErr}
+	}
+
 	cmc := defaultClusterMonitoringConfiguration()
-	err := UnmarshalStrict(content, &cmc)
+	err := k8syaml.UnmarshalStrict(content, &cmc)
```

Comment:
I see now.
I don't think we want to reintroduce `k8syaml.UnmarshalStrict`.
Let's just use the existing `err := UnmarshalStrict(content, &cmc)`, given that the prom-adapter config is still defined, the unmarshalling should pass.
After that, in a `func (c *Config) Precheck()` or somewhere else after unmarshalling, we can emit a warning if the field is set (in a follow-up PR as you're planning to do).

same for `NewUserConfigFromString`.